doc/sphinxext/gh_substitutions.py

```diff
@@ -0,0 +1,21 @@
+"""Provide a convenient way to link to GitHub issues and pull requests.
```

Comment:
Nice reflex @mathurinm to group the sphinx py modules in a separate folder.
We should consider doing the same for ``github_link.py`` (in another PR)